### DISCUSSION TITLE
Insert new row into external database

### DISCUSSION THREAD

**Discussion by thegrasshopper104** [Budibase/budibase#2701] [Help] [answered]
Hi everyone,
_first of all ysk that I'm a high school student and a beginner in...everything and English isn't my first language._ 

I'm trying to a**dd every newly created row to my MySQL database.** I connected to it and I got to automation and a new SQL query, but what do I write in it to include the data added to the budibase table?
(or alternatively, what's the best way to play youtube links from a budibase table during specific times?)
Thanks in advance.

**Comment by mjashanks** [comment, marked as answer]
Hey @aronkvh. Welcome! Firstly - "beginner in... everything" is a great place to be :) - always be learning!

**1. Create SQL row, from Budibase table **

We are currently working on improvements to automations, so this will become quite a bit easier after our next release (next week). However, the process for now is:

1. Create a new data source - a connection to your SQL DB
2. Add a custom query
3. Type out the full `INSERT` statement for your table. 
4. Add a parameter for every column in your table
5. in the `VALUES` part of your `INSERT`, feed in your parameters, like `INSERT INTO table ( column1 ) VALUES ( '{{ param1 }} ')`
6. Create your Budibase internal table
7. Create an automaton, with a "Row Created" trigger, from your Budibase table
8. Add an action for `Execute Data Connector` - this will allow you use your custom query.
9. Input the values into your query, from the trigger step e.g. `{{ trigger.row.Column1 }} `

https://user-images.githubusercontent.com/3524181/134341030-167b519d-b204-40f1-89ba-c146f50bbe67.mp4

**Playing YouTube links**

I'm not 100% what you are asking for - feel free to clarify. However, here is how you would
- Store YouTube links in a table
- Display the YouTube videos on a screen

1. Add your Budibase table (I'll call it "Videos"), with a column "Youtube URL"
2. In your Videos detail screen, create an `Embed` component
3. Input the youtube "Embed Code" into the component (you can find this on any YouTube video), it looks like:

`<iframe width="1244" height="700" src="https://www.youtube.com/embed/dQw4w9WgXcQ" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`

https://user-images.githubusercontent.com/3524181/134341861-0477fe57-dc80-43a9-b393-f632335a4f24.mp4

**Comment by thegrasshopper104** [comment]
Thanks a lot for the answer :)
What I was trying to do is play through the submitted links, and I just decided to export the table to csv and create a youtube playlist from it. 
but now I understand Budibase a bit more. (oh and it's nice to see that you are a Plasma user as well)
